### PR TITLE
Cleanup host discovery

### PR DESCRIPTION
**Pull request by aiordache**
Signed-off-by: Anca Iordache <anca.iordache@docker.com>
relates to https://github.com/moby/moby/issues/40383

### CODE REVIEW

**Comment by arkodg** [comment]
will review this once the moby counterpart PR is green since most swarm tests are present there

File: controller.go [hunk 1]
```diff
@@ -203,7 +200,7 @@ func New(cfgOptions ...config.Option) (NetworkController, error) {
 		return nil, err
 	}
 
-	drvRegistry, err := drvregistry.New(c.getStore(datastore.LocalScope), c.getStore(datastore.GlobalScope), c.RegisterDriver, nil, c.cfg.PluginGetter)
+	drvRegistry, err := drvregistry.New(c.getStore(datastore.LocalScope), c.getStore(datastore.GlobalScope), nil, nil, nil)
```
Comment (eunomie) [inline controller.go]: I think this change is not necessary and causes troubles. I have issues where tests in moby cannot start network plugins and I guess it comes from there and with change below

File: controller.go [hunk 2]
```diff
@@ -686,19 +637,6 @@ func (c *controller) GetPluginGetter() plugingetter.PluginGetter {
...
-func (c *controller) RegisterDriver(networkType string, driver driverapi.Driver, capability driverapi.Capability) error {
-	c.Lock()
-	hd := c.discovery
-	c.Unlock()
-
-	if hd != nil {
-		c.pushNodeDiscovery(driver, capability, hd.Fetch(), true)
-	}
-
-	c.agentDriverNotify(driver)
```
Comment (eunomie) [inline controller.go]: Shouldn't we keep this part? The lock and the block if `hd != nil` should be removed, but that line is independent from the node discovery I think.

**Comment by thaJeztah** [comment]
Thanks for looking into this; so I'd be interested to see if we still require `libkv` (and if we could get rid of the dependencies for external k/v stores). I see boltdb is also used through the k/v store (and _think_ we still use that, but not sure 😓). Here's a quick attempt to see what happens if we remove some of the code related to external k/v drivers; https://github.com/docker/libnetwork/pull/2521

**Comment by cpuguy83** [comment]
Note we have migrated this codebase over to github.com/moby/moby/libnetwork.
We are not accepting PR's on this repo anymore except for backports to be included in moby 20.10

**Comment by thaJeztah** [comment]
closing this one; this has been merged in moby through https://github.com/moby/moby/pull/42247

**Pull request closed without merging**